frontend/packages/model-registry/upstream/frontend/src/odh/hooks/useDeployModalExtension.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { HookNotify, useResolvedExtensions } from '@odh-dashboard/plugin-core';
+import { isModelRegistryDeployModalExtension } from '~/odh/extension-points';
+import MRDeployFormDataLoader from '~/odh/components/MRDeployFormDataLoader';
+import { ModelVersion } from '~/app/types';
+
+const useDeployModalExtension = ({
+  mv,
+  mvLoaded,
+  mvError,
+}: {
+  mv: ModelVersion;
+  mvLoaded: boolean;
+  mvError: Error | undefined;
+}) => {
+  const [extensions, extensionsLoaded] = useResolvedExtensions(isModelRegistryDeployModalExtension);
+  const hookStateActionProvider = React.useMemo(
+    () => extensionsLoaded && extensions?.[0]?.properties.useDeployButtonState,
+    [extensionsLoaded, extensions],
+  );
+  const ModalComponent = React.useMemo(
+    () => extensionsLoaded && extensions?.[0]?.properties.modalComponent,
+    [extensionsLoaded, extensions],
+  );
+
+  const [openModal, setOpenModal] = React.useState(false);
+
+  const [buttonState, setButtonState] = React.useState<{
+    visible: boolean;
+    enabled?: boolean;
+    tooltip?: string;
+  }>();
+
+  const hookNotifyComponent = React.useMemo(
+    () =>
+      hookStateActionProvider ? (
+        <HookNotify useHook={hookStateActionProvider} onNotify={(value) => setButtonState(value)} />
+      ) : null,
+    [hookStateActionProvider],
+  );
```

Comment:
It feels like maybe we should abstract this out into a `useHookNotify` hook so we can hide/encapsulate the extra `useState` that is just for holding the hook return value. It makes sense that we need to implement it this way but seeing that useState could be confusing since it's not really app state / it shouldn't be updated from anywhere outside that `onNotify` callback.

Not a blocker, but something to come back to probably.